### PR TITLE
Feature/add workplace cqc regulated no user account

### PR DESCRIPTION
**Pull request by omarCreativeDev**
- if workplace is cqc regulated route to create user account route
- relocate generation of addWorkplace request so it can be reused by 2 different components
- fix routes
- create WIP create user account component [actual component update to come in a separate pr]

### CODE REVIEW

**Comment by omarCreativeDev** [comment]
[(3) Parent and subsidiaries - Add a workplace (No user account)](https://trello.com/c/vuS39R5E/31-3-parent-and-subsidiaries-add-a-workplace-no-user-account)

File: src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts [hunk 1]
```diff
@@ -30,7 +28,7 @@ export class ConfirmWorkplaceDetailsComponent extends ConfirmWorkplaceDetails {
   }
 
   protected init(): void {
-    this.flow = '/registration';
+    this.flow = '/add-workplace';
```
Comment (ChazUK) [inline src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: Was this a mistake in previous PR?

Comment (omarCreativeDev) [reply src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: it was a previous mistake yup : (

File: src/app/core/services/workplace.service.ts [hunk 2]
```diff
@@ -26,4 +26,17 @@ export class WorkplaceService {
   public addWorkplace(establishmentuid: string, request: AddWorkplaceRequest): Observable<AddWorkplaceResponse> {
     return this.http.post<AddWorkplaceResponse>(`/api/establishment/${establishmentuid}`, request);
   }
+
+  public addWorkplaceRequest(locationAddress: LocationAddress, workplace: Service): AddWorkplaceRequest {
```
Comment (ChazUK) [inline src/app/core/services/workplace.service.ts]: Would prefer this to be generateAddWorkplaceRequest

File: src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts [hunk 3]
```diff
@@ -46,39 +44,38 @@ export class ConfirmWorkplaceDetailsComponent extends ConfirmWorkplaceDetails {
...
-      county: this.locationAddress.county,
-      isRegulated: this.workplace.isCQC,
-      locationName: this.locationAddress.locationName,
-      mainService: this.workplace.name,
-      postalCode: this.locationAddress.postalCode,
-      townCity: this.locationAddress.townCity,
-    };
+  public continue(): void {
+    if (this.workplace.isCQC) {
+      this.router.navigate([`${this.flow}/create-user-account`]);
```
Comment (ChazUK) [inline src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: I thought the user gets asked the question if they want to add a user account or not before being navigated into that flow?

File: src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts [hunk 4]
```diff
@@ -46,39 +44,38 @@ export class ConfirmWorkplaceDetailsComponent extends ConfirmWorkplaceDetails {
...
-      addressLine1: this.locationAddress.addressLine1,
-      addressLine2: this.locationAddress.addressLine2,
-      county: this.locationAddress.county,
-      isRegulated: this.workplace.isCQC,
-      locationName: this.locationAddress.locationName,
-      mainService: this.workplace.name,
-      postalCode: this.locationAddress.postalCode,
-      townCity: this.locationAddress.townCity,
-    };
+  public continue(): void {
```
Comment (ChazUK) [inline src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: Also, Can't remember if you were in the discussion but maybe check with Rich about where we want to save the workplace, if it's always after completing this step, or always on the final step.

File: src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts (hunk 3 above)
Comment (omarCreativeDev) [reply src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: yup you are correct, the `create-user-account` screen is basically the question if a user account is required or not and if yes then there will be another screen/s to follow

File: src/app/core/services/workplace.service.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/core/services/workplace.service.ts]: done

File: src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts (hunk 4 above)
Comment (omarCreativeDev) [reply src/app/features/add-workplace/confirm-workplace-details/confirm-workplace-details.component.ts]: as discuss we are on the same page about this, so make api on last step of the flow

**Pull request merged**